Bio/AlignIO/MafIO.py

```diff
@@ -0,0 +1,581 @@
+
+                # convert MAF-style +/- strand to biopython-type 1/-1
+                if line_split[4] == "+":
+                    strand = 1
+                elif line_split[4] == "-":
+                    strand = -1
+                else:
+                    strand = 1
+
+                # s (literal), src (ID), start, size, strand, srcSize, text (sequence)
+                anno = {"start": int(line_split[2]),
+                        "size": int(line_split[3]),
+                        "strand": strand,
+                        "srcSize": int(line_split[5])}
+
+                sequence = line_split[6]
+
+                # interpret a dot/period to mean same the first sequence
+                if "." in sequence:
+                    if not records:
+                        raise ValueError("Found dot/period in first sequence of alignment")
+
+                    ref = str(records[0].seq)
+                    new = []
+
+                    for (s, r) in zip(sequence, ref):
+                        new.append(r if s == "." else s)
+
+                    sequence = "".join(new)
+
+                records.append(SeqRecord(Seq(sequence, alphabet),
+                               id=line_split[1],
+                               name=line_split[1],
+                               description="",
+                               annotations=anno))
+            elif line.startswith("e") or \
+                 line.startswith("i") or \
+                 line.startswith("q"):
+                # not implemented
+                pass
+            elif not line.strip():
+                # end a bundle of records
+                if seq_count is not None:
+                    assert len(records) == seq_count
+
+                alignment = MultipleSeqAlignment(records, alphabet)
+                # TODO - Introduce an annotated alignment class?
+                # See also Bio/AlignIO/FastaIO.py for same requirement.
+                # For now, store the annotation a new private property:
+                alignment._annotations = annotations
+
+                yield alignment
+
+                in_a_bundle = False
+
+                annotations = []
+                records = []
+            else:
+                raise ValueError("Error parsing alignment - unexpected line:\n%s" % (line,))
+        elif line.startswith("a"):
+            # start a bundle of records
+            in_a_bundle = True
+
+            if len(line.strip().split()[1:]) != line.count("="):
+                raise ValueError("Error parsing alignment - invalid key in 'a' line")
+
+            annotations = dict([x.split("=") for x in line.strip().split()[1:]])
+        elif line.startswith("#"):
+            # ignore comments
+            pass
+        elif not line:
+            break
+
+
+class MafIndex(object):
+    def __init__(self, sqlite_file, maf_file, target_seqname):
+        """Indexes or loads the index of a MAF file"""
+
+        import os
```

Comment:
Certainly lets put ``import os`` at top level.

For SQLite3 there's the corner case of systems like Jython (and self-compiled Python missing library headers) where that import will fail, which we already have to handle for ``Bio.SeqIO.index`` etc.